features/web_api/wire_advanced_search.feature

```diff
@@ -0,0 +1,116 @@
+        Then we get the following order
+        """
+        []
+        """
+        When we get "/wire/search?advanced_search={"all":["Weather","Sydney"],"fields":["headline"]}"
+        Then we get the following order
+        """
+        ["weather-today-sydney"]
+        """
+        When we get "/wire/search?advanced_search={"all":["Weather","Sydney"],"fields":["slugline"]}"
+        Then we get the following order
+        """
+        []
+        """
+        When we get "/wire/search?advanced_search={"all":["Weather","Today"],"fields":["slugline"]}"
+        Then we get the following order
+        """
+        ["weather-today-sydney", "weather-today-prague", "weather-today-belgrade"]
+        """
+        When we get "/wire/search?advanced_search={"all":["Weather","Sydney","Report"],"fields":["body_html"]}"
+        Then we get the following order
+        """
+        ["weather-today-sydney"]
+        """
+
+    @auth @admin
+    Scenario: Search for any keywords
+        When we get "/wire/search?advanced_search={"any":["Sydney","Prague","Belgrade","Wellington"]}"
+        Then we get the following order
+        """
+        ["weather-today-sydney", "weather-today-prague", "weather-today-belgrade"]
+        """
+        When we get "/wire/search?advanced_search={"any":["Wellington","Canberra"]}"
+        Then we get the following order
+        """
+        []
+        """
+        When we get "/wire/search?advanced_search={"any":["Sydney","Prague","Belgrade","Wellington"],"fields":["headline"]}"
+        Then we get the following order
+        """
+        ["weather-today-sydney", "weather-today-prague", "weather-today-belgrade"]
+        """
+        When we get "/wire/search?advanced_search={"any":["Sydney","Prague","Belgrade","Wellington"],"fields":["slugline"]}"
+        Then we get the following order
+        """
+        []
+        """
+        When we get "/wire/search?advanced_search={"any":["Sydney","Prague","Belgrade","Wellington"],"fields":["body_html"]}"
+        Then we get the following order
+        """
+        ["weather-today-sydney", "weather-today-prague", "weather-today-belgrade"]
+        """
+
+    @auth @admin
+    Scenario: Search for excluded keywords
+        When we get "/wire/search?advanced_search={"exclude":["Belgrade","Wellington","Canberra"]}"
+        Then we get the following order
+        """
+        ["weather-today-sydney", "weather-today-prague", "sports-results-today-1"]
+        """
+        When we get "/wire/search?advanced_search={"exclude":["Weather"]}"
+        Then we get the following order
+        """
+        ["sports-results-today-1"]
+        """
+        When we get "/wire/search?advanced_search={"exclude":["Weather"],"fields":["headline"]}"
+        Then we get the following order
+        """
+        ["sports-results-today-1"]
+        """
+
+    @auth @admin @skip
+    Scenario: Search multiple fields
```

Comment:
Using ES validate & explain, I got the following:

Slugline & Headline:
```
+(
    +blended(terms:[slugline:sydney, headline:sydney])
    +blended(terms:[slugline:current, headline:current])
    +blended(terms:[slugline:report, headline:report])
)
```

Headline & Body HTML
```
+(
    (+body_html:sydney +body_html:current +body_html:report) |
    (+headline:sydney +headline:current +headline:report)
)
```

As you can see, the `headline & body_html` does not work as intended

Reply:
ok so we need to tweak the mapping right? it would be useful to control it in newsroom-core repo anyhow, so we might register an items resource for this explicitly here rather than doing the changes only in sd-core